packages/stream-model/src/model.ts

```diff
@@ -19,7 +19,7 @@ import { create } from 'multiformats/hashes/digest'
 import { code, encode } from '@ipld/dag-cbor'
 import multihashes from 'multihashes'
 
-export const MODEL_VERSION_REGEXP = /^[0-9]+\.[0-9]+$/
+export const MODEL_VERSION_REGEXP = /^[0-9]+\.[0-9]+(\.[0-9]+)?/
```

Comment:
I think we should just use a semver library here to extract an actual version. As for this change, it has no impact, since we only check model definition language version for major/minor capability. This change just makes the model version check more permissive for other model implementors, so they can specify `1.0` or `1.0.0` and still have it work.

Reply:
Why adding a semver library please? We don't need something as complex as semver, so I'd rather avoid adding extra dependencies that are not needed.
Yes the logic just checks for major/minor so what is the point of adding extra information that's not useful? It's still going to leak for example if we want to produce stats counting the number of models in different versions, we'd need to take into account that `1.0.x` needs to be considered as `1.0`, for not other reason that we introduced this change.

Reply:
Main reason is to follow the robustness principle/Postel's law. People communicating with composedb should not have to know that the version we expect is _only_ major.minor, only that they need to pass in a version, and we will then determine if it is valid.

If we then want to count stats,  we count based on the tuple of (major, minor) not on the version passed in.

Reply:
To me it's the same rationale as Spence's comment on the model stream ID: the `version` field is part of the model protocol and it should be restrictive. I don't see the point of allowing models with a version of `1.0.999-yolo` that doesn't mean anything for the protocol just because we can.